amethyst_assets/src/store/dir.rs

```diff
@@ -33,34 +34,66 @@ impl Directory {
 impl Store for Directory {
     type Error = IoError;
 
-    fn modified(&self, category: &str, id: &str, ext: &str) -> Result<u64, IoError> {
-        use std::fs::metadata;
-
+    fn modified(&self, category: &str, id: &str, exts: &[&str]) -> Result<u64, IoError> {
         let mut path = self.loc.clone();
 
         path.push(category);
         path.push(id);
-        path.set_extension(ext);
 
-        Ok(metadata(&path)?.modified()?.duration_since(UNIX_EPOCH).unwrap().as_secs())
+        for ext in exts {
+            match fs::metadata(path.with_extension(ext)) {
+                Ok(meta) => {
+                    return Ok(
+                        meta.modified()?
+                            .duration_since(UNIX_EPOCH)
+                            .unwrap()
+                            .as_secs(),
+                    );
+                }
+                Err(_) => (),
```

Comment:
The new `modified` method could return metadata for `file.otherext` if the user lacks permission to access the matadata of `file.correctext`'.